File: push/src/main/java/com/example/push/MainActivity.java
package com.example.push;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;

import com.umeng.message.PushAgent;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        if (true) {//"hasAgreementAgreed"
            PushAgent.getInstance(this).onAppStart();
        }
    }
}

File: react_native_sample/src/main/java/com/example/react_native_sample/MainActivity.kt
package com.example.react_native_sample

import android.os.Bundle
import android.widget.Button
import androidx.appcompat.app.AppCompatActivity
import com.facebook.react.ReactApplication
import com.facebook.react.ReactInstanceManager


class MainActivity : AppCompatActivity() {


//    private val mReactInstanceManager: ReactInstanceManager by lazy {
//        (application as ReactApplication).reactNativeHost.reactInstanceManager
//    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

//        mReactInstanceManager.createReactContextInBackground()

        findViewById<Button>(R.id.btn).setOnClickListener {
            ExtendReactActivity.launch(this@MainActivity)
        }
    }

    override fun onDestroy() {
        super.onDestroy()
//        mReactInstanceManager.onHostDestroy(this)
    }
}